### PR TITLE
Add a Note for Including/Excluding files

### PR DESCRIPTION
**Pull request by sugersuger**
**Add a Note of `AllCops` / `Exclude`**  

I think that It would be helpful to document cases where the `AllCops`/`Exclude` rules do not apply.
[rubocop-hq/ruby-style-guide#2492 (comment)](https://github.com/rubocop-hq/rubocop/issues/2492#issuecomment-164375887)

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [ ] Squashed related commits together.
* [ ] Added tests.
* [ ] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by bbatsov** [comment]
I think that's hard to understand without showing some examples.

**Comment by sugersuger** [comment]
@bbatsov  
Thank you for your review.
Added an example.

File: manual/configuration.md [hunk 1]
```diff
@@ -271,6 +271,31 @@ AllCops:
 # ...
 ```
 
+**Note**: When inspecting a certain directory(or file)
+given as Rubocop's command line arguments,
```
Comment (koic) [inline manual/configuration.md]: The spelling of the name is `RuboCop` :-)

File: manual/configuration.md [hunk 2]
```diff
@@ -271,6 +271,31 @@ AllCops:
...
+# .rubocop.yml
+AllCops:
+  Exclude:
+    - foo.rb
+```
+
+If `foo.rb` specified as a Rubocop's command line argument, the result is:
+
+```sh
+# The RuboCop inspects foo.rb.
```
Comment (bbatsov) [inline manual/configuration.md]: Remove the `The` here.

File: manual/configuration.md [hunk 3]
```diff
@@ -271,6 +271,31 @@ AllCops:
...
+$ bundle exec rubocop foo.rb
+
+# The RuboCop does not inspect foo.rb.
```
Comment (bbatsov) [inline manual/configuration.md]: Same here.

File: manual/configuration.md (hunk 1 above)
Comment (sugersuger) [reply manual/configuration.md]: Thank you for point out my mistake!
I corrected it.

File: manual/configuration.md (hunk 2 above)
Comment (sugersuger) [reply manual/configuration.md]: I removed it.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**